src/snowflake/snowpark/mock/_functions.py

```diff
@@ -523,26 +608,47 @@ def mock_to_timestamp(
 
         [ ] If the value is greater than or equal to 31536000000000000, then the value is treated as nanoseconds.
     """
+    # dateutil is a pandas dependency
+    import dateutil.parser
+
     res = []
-    auto_detect = bool(not fmt)
-    default_format = "%Y-%m-%d %H:%M:%S.%f"
-    (
-        timestamp_format,
-        hour_delta,
-        fractional_seconds,
-    ) = convert_snowflake_datetime_format(fmt, default_format=default_format)
+    fmt_column = fmt if fmt is not None else [None] * len(column)
+
+    for data, format in zip(column, fmt_column):
```

Comment:
is `fmt_column` guaranteed to be a list of the same length as column?
I'm wondering if `fmt_column` is just a string literal, then zip only contains the one data

Reply:
I'm not sure if it was at the time of your comment, but it should be now.